msg/output_control.msg

```diff
@@ -0,0 +1,81 @@
+uint64 timestamp         # time since system start (microseconds)
+uint64 timestamp_sample  # the timestamp the data this control response is based on was sampled
+
+uint8 NUM_OUTPUT_CONTROL_GROUPS = 5 # Max number of instances of this topic
+uint8 MAX_ACTUATORS = 16            # Max actuators available in a single topic publication
+
+float32[16] value         # Normalized control value in range [-1, 1]
+
+# Actuator Function List
+uint16 FUNCTION_NONE  = 0  # No output
+uint16 FUNCTION_MIXER = 1  # Output comes from mixer file
+
+# Misc. functions for different vehicle types
+# You must implement a separate 'TOPICS' instance below for each of these you wish to use!
+uint16 FUNCTION_PARACHUTE         = 10
+uint16 FUNCTION_LANDING_GEAR      = 11
+uint16 FUNCTION_LANDING_GEAR_DOOR = 12
+uint16 FUNCTION_FLAPS             = 13
+uint16 FUNCTION_SPOILER           = 14
+uint16 FUNCTION_AIRBRAKS          = 15
+uint16 FUNCTION_PAYLOAD_RELEASE   = 16
+
+# Up to 8 individual actuators controlled by DO_SET_{SERVO,ACTUATOR}
+uint16 FUNCTION_MAVLINK_SERVO0 = 100
+uint16 FUNCTION_MAVLINK_SERVO1 = 101
+uint16 FUNCTION_MAVLINK_SERVO2 = 102
+uint16 FUNCTION_MAVLINK_SERVO3 = 103
+uint16 FUNCTION_MAVLINK_SERVO4 = 104
+uint16 FUNCTION_MAVLINK_SERVO5 = 105
+uint16 FUNCTION_MAVLINK_SERVO6 = 106
+uint16 FUNCTION_MAVLINK_SERVO7 = 107
+
+# Specific multicopter functions
+# This allows you to easily write your own mixer for multicopters
+uint16 FUNCTION_MC_MOTOR1 = 200
+uint16 FUNCTION_MC_MOTOR2 = 201
+uint16 FUNCTION_MC_MOTOR3 = 202
+uint16 FUNCTION_MC_MOTOR4 = 203
+uint16 FUNCTION_MC_MOTOR5 = 204
+uint16 FUNCTION_MC_MOTOR6 = 205
+uint16 FUNCTION_MC_MOTOR7 = 206
+uint16 FUNCTION_MC_MOTOR8 = 207
```

Comment:
I'm actually actively testing this right now ;)
Still open to debate, but this is the topic I'm using for CAN-->PWM control of ESCs using UAVCANv1.  the FW topic will be used to control the FW servos + engine(s) alongside the MC motors.

Reply:
Cool. What do you mean by CAN->PWM bridge, do you have an external CAN node sending setpoints and PX4 just doing conversion to PWM ?

Reply:
Sort of - using PX4 as a distributed system; the core FC sends out the UAVCAN setpoint messages, then to avoid running a whole bunch of PWM lines throughout the vehicle, you use a small PX4-based CANnode at the actuator that converts the UAVCAN setpoint to a PWM value and sends it to the (non-UAVCAN-enabled) actuator.

That's the use case that kicked off my interest in this entire topic :)

Reply:
That is a cool use case. 
I think the name should indeed reflect that (i) this is UAVCAN-related and (ii) this is coming from an external system. What about output_control_ext_can (not convinced myself) ?

Another question, is there a distinction between flap and rotor at the UAVCAN level that justifies having separate topics for mc and fw?

Reply:
> I think the name should indeed reflect that (i) this is UAVCAN-related and (ii) this is coming from an external system. What about output_control_ext_can (not convinced myself) ?

Yeah, that would probably be a better choice... but I'd kinda prefer to just keep things as-is for the moment and merge it in so that others like @bkueng can start thinking about how it fits into the rest of the system, and iterate / refine once we have more feedback from different perspectives.  A name change would be a simple PR - but I like the idea of keeping the "_mc" and "_fw" topics in initially, just as a sort of guide as to what is possible as this gets developed further.

> Another question, is there a distinction between flap and rotor at the UAVCAN level that justifies having separate topics for mc and fw?

There is, actually - see:
https://github.com/UAVCAN/public_regulated_data_types/blob/master/reg/drone/service/actuator/esc/_.0.1.uavcan
and
https://github.com/UAVCAN/public_regulated_data_types/blob/master/reg/drone/service/actuator/servo/_.0.1.uavcan

It's not a big difference, but there are benefits to handling servos separately from motors.

Reply:
Sounds good, let's merge this.